tensorflow/contrib/cmake/CMakeLists.txt

```diff
@@ -467,6 +467,10 @@ if (tensorflow_ENABLE_GPU)
   include_directories(${tensorflow_source_dir}/third_party/gpus)
   # add cuda libraries to tensorflow_EXTERNAL_LIBRARIES
   list(APPEND tensorflow_EXTERNAL_LIBRARIES ${CUDA_LIBRARIES})
+  if(NOT WIN32)
+    # add gomp to tensorflow_EXTERNAL_LIBRARIES, needed by libcusolver.so
+    list(APPEND tensorflow_EXTERNAL_LIBRARIES gomp)
```

Comment:
I'm not sure if this is the right way to do it.

On Ubuntu 16.04, I indeed had a file `/usr/lib/x86_64-linux-gnu/libgomp.so`, but on 14.04 and 18.04, I only have `/usr/lib/x86_64-linux-gnu/libgomp.so.1`, which is not found by the linker. There are several `libgomp.so`s in subdirectories for each installed gcc version, e.g. `/usr/lib/gcc/x86_64-linux-gnu/6/libgomp.so`. These subdirectories are not however on the linker path, so linking a dependent project fails.

What works for me (and IMHO is the correct way) is `find_package(OpenMP)` and then exporting `OpenMP_CXX_FLAGS` to dependent projects (using an extras file?). With gcc, it adds `-fopenmp` to both compile and link steps. 

In https://github.com/tensorflow/tensorflow/commit/9e197152c04ebb81f055067534bd93322d182f0e there was some alteration of the CXX flags done (just for the current project, not for dependent ones), but as stated in the discussion to https://github.com/tensorflow/tensorflow/pull/18973 , it'd be better to find the flags using the `find_package` call. This is just a reminder that the work should be coordinated.

Reply:
@peci1 I only tested on Ubuntu 16.04 as it was the version Nvidia support for CUDA (Nvidia supports ubuntu 16.04 and 17.04 at that time, I couldn't find 14.04 or 18.04), but let me take a look at Ubuntu 18.04 and see if it is possible to address the issue you mentioned.

Reply:
Can't seem to find much documentation on support for cmake on Ubuntu 18.04.

Reply:
@aceveggie I haven't been able to find the cuda install on ubuntu 18.04, or the cuda docker images based on Ubuntu 18.04: https://hub.docker.com/r/nvidia/cuda/

I can look into fixing any potential issues of CMAKE+CUDA once Nvidia supports 18.04.

Reply:
Installing the 17.04 or 17.10 binary packages works on 18.04 without a problem (I've built TF both with bazel and CMake on 18.04 with CUDA 9.0 and 9.1). But I understand it still lacks official support.